test/server.cpp

```diff
@@ -1176,3 +1190,69 @@ TEST_F(LibraryServerTest, catalog_v2_entries_filtered_by_search_terms)
     "</feed>\n"
   );
 }
+
+TEST_F(LibraryServerTest, catalog_v2_inidividual_entry_access)
+{
+  const auto r = zfs1_->GET("/catalog/v2/entries/raycharles");
+  EXPECT_EQ(r->status, 200);
+  EXPECT_EQ(maskVariableOPDSFeedData(r->body),
+    "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n"
+    "<feed xmlns=\"http://www.w3.org/2005/Atom\"\n"
+    "      xmlns:opds=\"https://specs.opds.io/opds-1.2\"\n"
+    "      xmlns:opensearch=\"http://a9.com/-/spec/opensearch/1.1/\">\n"
+    "  <id>12345678-90ab-cdef-1234-567890abcdef</id>\n"
+    "\n"
+    "  <link rel=\"self\"\n"
+    "        href=\"/catalog/v2/entries/raycharles\"\n"
+    "        type=\"application/atom+xml;profile=opds-catalog;kind=acquisition\"/>\n"
+    "  <link rel=\"start\"\n"
+    "        href=\"/catalog/v2/root.xml\"\n"
+    "        type=\"application/atom+xml;profile=opds-catalog;kind=navigation\"/>\n"
+    "  <link rel=\"up\"\n"
+    "        href=\"/catalog/v2/entries\"\n"
+    "        type=\"application/atom+xml;profile=opds-catalog;kind=acquisition\"/>\n"
+    "\n"
+    "  <title>Single Entry</title>\n"
+    "  <updated>YYYY-MM-DDThh:mm:ssZ</updated>\n"
+    "\n"
+    RAY_CHARLES_CATALOG_ENTRY
+    "</feed>\n"
+  );
+
+  const auto r1 = zfs1_->GET("/catalog/v2/entries/non-existent-entry");
+  EXPECT_EQ(r1->status, 404);
+}
+
+TEST_F(LibraryServerTest, catalog_v2_partial_entries)
+{
+  const auto r = zfs1_->GET("/catalog/v2/partial_entries");
+  EXPECT_EQ(r->status, 200);
+  EXPECT_EQ(maskVariableOPDSFeedData(r->body),
+    CATALOG_V2_PARTIAL_ENTRIES_PREAMBLE("")
+    "  <title>All Entries</title>\n"
+    "  <updated>YYYY-MM-DDThh:mm:ssZ</updated>\n"
+    "\n"
+    "  <entry>\n"
+    "    <id>urn:uuid:charlesray</id>\n"
+    "    <link rel=\"alternate\"\n"
+    "          href=\"/catalog/v2/entries/charlesray\"\n"
+    "          type=\"application/atom+xml;type=entry;profile=opds-catalog\"\n"
+    "          title=\"Charles, Ray\"/>\n"
+    "  </entry>\n"
+    "  <entry>\n"
+    "    <id>urn:uuid:raycharles</id>\n"
+    "    <link rel=\"alternate\"\n"
+    "          href=\"/catalog/v2/entries/raycharles\"\n"
+    "          type=\"application/atom+xml;type=entry;profile=opds-catalog\"\n"
+    "          title=\"Ray Charles\"/>\n"
+    "  </entry>\n"
+    "  <entry>\n"
+    "    <id>urn:uuid:raycharles_uncategorized</id>\n"
+    "    <link rel=\"alternate\"\n"
+    "          href=\"/catalog/v2/entries/raycharles_uncategorized\"\n"
+    "          type=\"application/atom+xml;type=entry;profile=opds-catalog\"\n"
+    "          title=\"Ray (uncategorized) Charles\"/>\n"
```

Comment:
I'm a bit disturb by this attribute here, I would have put the title as a node in the `entry`.
But it is what we have in the example in the spec 🤷🏽‍♂️

Reply:
Added the `<title>` node without removing the `title` attribute from the link node. In the example, the latter is used to make it explicit that the link leads to the complete entry (note the word "Complete" in the title). I think it's safe to drop the title attribute from `<link rel="alternate">`

Reply:
Yes, we can remove the title from the link. It is a duplicated value and it is better in a `<title>` node.